test/mocks/MockHttp.scala

```diff
@@ -48,6 +48,16 @@ trait MockHttp extends AnyWordSpecLike with Matchers with OptionValues with Befo
     ).thenReturn(Future.successful(response))
   }
 
+  def mockIFGet[A, B](url: String, queryParameters: Seq[(String, String)], headers: Seq[(String, String)])(response: Either[A, B]): Unit = {
```

Comment:
Could we de-duplicate mockIfGet() and mockDesGet()? We could even replace both of these with a mockGet() call since they are identical?